pkg/sql/resolver.go

```diff
@@ -703,6 +707,46 @@ func newInternalLookupCtx(descs []catalog.Descriptor, prefix *dbdesc.Immutable)
 			schemaDescs[desc.GetID()] = desc
 		}
 	}
+
+	if len(typDescs) > 0 {
```

Comment:
is there a way to reuse this code with the logic done in the descriptor collection?

Reply:
Played around with a few ways to factor it out. Ended up with a `HydrateDescriptors` in `descs`. Curious what you think of it.